libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -235,20 +235,21 @@ void NavEKF3_core::writeOptFlowMeas(uint8_t &rawFlowQuality, Vector2f &rawFlowRa
 // check for new magnetometer data and update store measurements if available
 void NavEKF3_core::readMagData()
 {
-    if (!_ahrs->get_compass()) {
+    const Compass &compass = AP::compass();
```

Comment:
Nitpicking, but can we make EKF2 and EKF3 be consistent? Looks like EKF2 code is after the comment below, EKF3 code is before the comment - I prefer the EKF3 way, but I just care that changes are consistent.

Reply:
OK